content/docs/user-guide/experiment-management/checkpoints.md

```diff
@@ -0,0 +1,456 @@
+
+```yaml
+stages:
+  download:
+    cmd: python download.py
+    deps:
+      - download.py
+    outs:
+      - data/MNIST
+  train:
+    cmd: python train.py
+    deps:
+      - data/MNIST
+      - train.py
+    params:
+      - seed
+      - lr
+      - weight_decay
+    outs:
+      - model.pt:
+          checkpoint: true
+    plots:
+      - predictions.json:
+          cache: false
+    live:
+      dvclive:
+        summary: true
+        html: true
+```
+
+Before we go any further, this is a great point to add these changes to your Git
+history. You can do that with the following commands:
+
+```bash
+git add .
+git commit -m "created DVC pipeline"
+```
+
+Now that you know how to enable checkpoints in a DVC pipeline, let's move on to
+setting up checkpoints in your code.
+
+## Registering checkpoints in your code
+
+Take a look at the _train.py_ file and you'll see how we train a convolutional
+neural network to classify handwritten digits. The main area of this code most
+relevant to checkpoints is when we iterate over the training epochs.
+
+This is where DVC will be able to track our metrics over time and where we will
+add our checkpoints to give us the points in time with our model that we can
+switch between.
+
+Now we need to enable checkpoints at the code level. We are interested in
+tracking the metrics along with each checkpoint, so we'll need to add a few
+lines of code.
+
+In the `train.py` file, import the `dvclive` package with the other imports:
+
+```bash
+import dvclive
+```
+
+Then update the following lines of code in the `main` method inside of the
+training epoch loop.
+
+```python
+# Iterate over training epochs.
+for i in range(1, EPOCHS+1):
+    # Train in batches.
+    train_loader = torch.utils.data.DataLoader(
+            dataset=list(zip(x_train, y_train)),
+            batch_size=512,
+            shuffle=True)
+    for x_batch, y_batch in train_loader:
+        train(model, x_batch, y_batch, params["lr"], params["weight_decay"])
+    torch.save(model.state_dict(), "model.pt")
+    # Evaluate and checkpoint.
+    metrics = evaluate(model, x_test, y_test)
+    for k, v in metrics.items():
+-        print('Epoch %s: %s=%s'%(i, k, v))
++        dvclive.log(k, v)
++    dvclive.next_step()
+```
+
+The line `torch.save(model.state_dict(), "model.pt")` updates the checkpoint
+file.
+
+The `dvclive.log(k, v)` line stores the metric _k_ with a value _v_ in plain
+text files in the _dvclive_ directory by default.
+
+The `dvclive.next_step()` line tells DVC that it can take a snapshot of the
+entire workspace and version it with Git. It's important that with this approach
+only code with metadata is versioned in Git (as an ephemeral commit), while the
+actual model weight file will be stored in the DVC data cache.
+
+## Running experiments
+
+With checkpoints enabled and working in our code, let's run the experiment. You
+can run an experiment with the following command:
+
+```bash
+dvc exp run
```

Comment:
Please use the `dvc` highlighter (instead of `bash`) and `$` prompts throughout. Here's one:
````suggestion
```dvc
$ dvc exp run
````